python/ShapeTools.py

```diff
@@ -65,16 +69,63 @@ def doIndividualModels(self):
                 pdf.setStringAttribute("combine.process", p)
                 pdf.setStringAttribute("combine.channel", b)
                 pdf.setAttribute("combine.signal", self.DC.isSignal[p])
+                if self.DC.isSignal[p]:
+                    pdf.setAttribute('skipForErrorSum')
```

Comment:
The automatic bin-by-bin creation below may have to decide when to transition a bin from a single gaussian parameter, to per-process poisson parameters. It does this by summing the nominal yields and errors of all processes, except those marked with the attribute "skipForErrorSum". By default we apply this to all signals, since the initial signal normalisation is usually arbitrary and we don't want an unreasonably high initial signal rate to push us into the poisson regime unnecessarily. However we probably need a way to give the user some control over this.